aws-sigv4/src/lib.rs

```diff
@@ -14,24 +14,62 @@ pub mod types;
 
 type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
 
+use http::header::HeaderName;
 use sign::{calculate_signature, encode_with_hex, generate_signing_key};
 use types::{AsSigV4, CanonicalRequest, DateTimeExt, StringToSign};
+use std::time::{Instant, UNIX_EPOCH, SystemTime};
 
 pub fn sign<B>(
     req: &mut http::Request<B>,
     credential: &Credentials,
     region: &str,
     svc: &str,
 ) -> Result<(), Error>
-where
-    B: AsRef<[u8]>,
+    where
+        B: AsRef<[u8]>,
 {
+    for (header_name, header_value) in sign_core(&req, req.body(), credential, region, svc, SystemTime::now()) {
```

Comment:
`SystemTime::now()` isn't monotonic. Could that have an impact on signing? I'm guessing "no", given distributed systems-and-all.